pkg/azure/blob.go

```diff
@@ -180,14 +180,25 @@ func (c *BlobClient) BucketExists(ctx context.Context, bucketName string) (bool,
 	if err != nil {
 		return false, err
 	}
-	_, err = container.GetProperties(ctx, nil)
-	if err != nil {
-		var stgErr *azblob.StorageError
-		if errors.As(err, &stgErr) {
-			if stgErr.ErrorCode == azblob.StorageErrorCodeContainerNotFound {
+
+	var max int32 = 1
+	items := container.ListBlobsFlat(&azblob.ContainerListBlobsFlatOptions{
+		MaxResults: &max,
+	})
```

Comment:
The Azure SDK has a helper for this, see: https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/to/to.go#L10

Given this, I would likely replace this with:
```suggestion
	items := container.ListBlobsFlat(&azblob.ContainerListBlobsFlatOptions{
		MaxResults: to.Ptr(1),
	})
```